### ISSUE TITLE
Having a way to apply string/text transformation like `replace`

### ISSUE THREAD

**Issue by mikefaille** [dhall-lang/dhall-haskell#1973]
As a usecase, I use dhall-kubernetes and I want to generate my config from a simple input. 

As an example, I want use the domain name `example.com` to populate my config.

But, I have a limitation due to kubernetes. I can't reuse the `.` (dot) char on `namespace`. Then, normally, with a configuration tool like `helm` I replace `.` with `-` to get a `namespace` named like `example-com`.

At the end, creating new strings by replacing char from a previous one is a go/no-go for me to use dhall (vs jsonnet). I'm not a pro in haskell even if I try. I will need assistance to make it work or community contribution. For sure, we can simply position this issue as a _feature request._ Thank you for any help ! 

PS : Sorry for my english.

**Comment by Gabriel439** [comment]
@mikefaille: Yeah, this has been requested on several occasions and I'd be fine adding language support for a `Text/replace` utility

An example of another context where this came up is:

https://discourse.dhall-lang.org/t/text-manipulation-functions/176/3

I think the rough rule of thumb we can go with for now is to permit `Text` transformation functions that don't unlock introspection, so `Text/replace` is fine in that regard

**Comment by sjakobi** [comment]
I'll move this to the `dhall-lang` repo, where the standard lives and most language features are discussed.
